core/src/main/java/com/dtolabs/rundeck/core/jobs/JobService.java

```diff
@@ -119,6 +119,27 @@
     String startJob(JobReference jobReference, String jobArgString, String jobFilter, String asUser)
         throws JobNotFound;
 
+    /**
+     * Run a job
+     * @param jobReference reference to a job
+     * @param executionType type of invocation
+     * @param provenance    invocation provenance information
+     * @param jobArgString argString for the execution
+     * @param jobFilter filter for the execution
+     * @param asUser user to execute the job(null for the same user)
+     * @return Id of the result execution
+     * @throws JobNotFound if the specified job was not found
+     * @throws JobExecutionError if an error occurred executing the job
+     */
+    ExecutionReference runJob(
+            JobReference jobReference,
+            String executionType,
+            Map<String, String> provenance,
```

Comment:
Would it make sense to change these lines to `Provenance provenance` since the single provenance object has both pieces of information and as a type is more explicit?

Reply:
Yes it would be better